ratis-server/src/test/java/org/apache/ratis/server/impl/RaftReconfigurationBaseTest.java

```diff
@@ -145,6 +146,54 @@ public void testAddPeers() throws Exception {
     });
   }
 
+  /**
+   * Test leader election when changing cluster from single mode to HA mode.
+   */
+  @Test
+  public void testLeaderElectionWhenChangeFromSingleToHA() throws Exception {
+    runWithNewCluster(1, cluster -> {
+      RaftTestUtil.waitForLeader(cluster);
+
+      RaftGroupId groupId = cluster.getGroup().getGroupId();
+      RaftPeer curPeer = cluster.getGroup().getPeers().iterator().next();
+      RaftPeer newPeer = cluster.addNewPeers(1, true, true).newPeers[0];
+
+      RaftServerProxy leaderServer = cluster.getServer(curPeer.getId());
+
+      // Update leader conf to transitional single mode.
+      RaftConfigurationImpl oldNewConf = RaftConfigurationImpl.newBuilder()
+          .setOldConf(new PeerConfiguration(Arrays.asList(curPeer)))
+          .setConf(new PeerConfiguration(Arrays.asList(curPeer, newPeer)))
+          .setLogEntryIndex(Long.MAX_VALUE / 2)
+          .build();
+      Assert.assertTrue(oldNewConf.isSingleMode(curPeer.getId()));
+      leaderServer.setRaftConf(groupId, oldNewConf);
+      leaderServer.triggerElection(groupId);
```

Comment:
Use `transferLeadership` with null new leader and then remove the new `triggerElection` method.
```java
      try(RaftClient client = cluster.createClient()) {
        client.admin().transferLeadership(null, leaderServer.getId(), 10_000);
      }
```